src/flint/fmpz.jl

```diff
@@ -2569,8 +2569,8 @@ function add!(a::ZZRingElem, b::ZZRingElem, c::Ptr{Int})
   return a
 end
 
-add!(z::ZZRingElem, a::ZZRingElem, b::Integer) = add!(z, a, ZZRingElem(b))
-add!(z::ZZRingElem, x::Int, y::ZZRingElem) = add!(z, y, x)
+add!(z::ZZRingElemOrPtr, a::ZZRingElemOrPtr, b::Integer) = add!(z, a, flintify(b))
+add!(z::ZZRingElemOrPtr, x::Int, y::ZZRingElemOrPtr) = add!(z, y, x)
```

Comment:
This overlaps with PR #1873 so here I also added the OrPtr, and in the other PR the "flintify", hoping that this might avoid or at least minimize merge conflicts.